docs/common-github-actions-practices.md

```diff
@@ -336,40 +378,322 @@ This is what we use in the OpenTelemetry Java repositories:
 
 ### Prepare release branch
 
-The specifics depend a lot on the build tool and your version bumping needs.
+The specifics depend a lot on your specific version bumping needs.
+
+For OpenTelemetry Java repositories, the version in the `main` branch always ends with `-SNAPSHOT`,
+so preparing the release branch involves
+
+* removing `-SNAPSHOT` from the version on the release branch
+  (e.g. updating the version from `1.2.0-SNAPSHOT` to `1.2.0`)
+* bumping the version to the next `-SNAPSHOT` on main
+  (e.g. updating the version from `1.2.0-SNAPSHOT` to `1.3.0-SNAPSHOT`)
+
+```
```

Comment:
Nit: can we specify the language of the code samples in this file? Makes them a bit easier to read in GH
```suggestion
```yaml
```